### PR TITLE
Add fix for deploy action failure

### PR DESCRIPTION
**Pull request by abrennan89**
One of my manual cherrypicked PRs has a check failing for deploying the preview: https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/actions/runs/8648296395/job/23711771598?pr=161

image

It seems to be happening occassionally for other PRs also, so I tried to look into a fix for it. Based on https://github.com/peaceiris/actions-gh-pages/issues/339 I am proposing that updating the variable here from `github_token` to `deploy_key` may fix the issue, but I am not an expert on GH actions, so will need someone to sanity check 🙂

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -81,19 +81,19 @@ jobs:
...
+          deploy_key: ${{ secrets.RHDH_BOT_TOKEN }}
           publish_branch: gh-pages
           keep_files: true
           publish_dir: ./titles-generated
 
       - name: PR comment with doc preview
         uses: actions/github-script@v7
         with:
-          github-token: ${{ secrets.RHDH_BOT_TOKEN }}
+          deploy_key: ${{ secrets.RHDH_BOT_TOKEN }}
```
Comment (nickboldt) [inline .github/workflows/pr.yml:92]:
```suggestion
          github-token: ${{ secrets.RHDH_BOT_TOKEN }}
```

See https://github.com/actions/github-script?tab=readme-ov-file#using-a-separate-github-token -- deploy_token is not supported for this action.

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -81,7 +81,7 @@ jobs:
         uses: peaceiris/actions-gh-pages@v3
         # if: github.ref == 'refs/heads/main'
         with:
-          github_token: ${{ secrets.RHDH_BOT_TOKEN }}
+          deploy_key: ${{ secrets.RHDH_BOT_TOKEN }}
```
Comment (nickboldt) [inline .github/workflows/pr.yml:84]: we can try this if you want -- but I'm not sure if we're using https or ssh to check out sources.

https://github.com/peaceiris/actions-gh-pages?tab=readme-ov-file#supported-tokens

Comment (abrennan89) [reply .github/workflows/pr.yml:84]: Ah, I see. Is there a way to check?

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -81,7 +81,7 @@ jobs:
         uses: peaceiris/actions-gh-pages@v3
```
Comment (nickboldt) [inline .github/workflows/pr.yml:81]:
```suggestion
        uses: peaceiris/actions-gh-pages@v4
```

File: .github/workflows/pr.yml (hunk 2 above)
Comment (nickboldt) [reply .github/workflows/pr.yml:84]: try it and see if it breaks; then revert if so :D

**Pull request closed without merging**